modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/util/MBMessageUtil.java

```diff
@@ -71,6 +133,103 @@ public static List<MBMessage> getThreadMessages(
 		return messages;
 	}
 
+	public static String renderDifferentLevelMessages(
```

Comment:
The "differentLevel" part sounds weird. Can you refactor this so that it takes the message you're notifying about, move the _getParentMessages in the other class here (and make it private), pass the maximum number of messages and rename this method to "renderMBMessageInContext".

Reply:
Also, move the logic that calls this and the "sameLevel" method here, and hide everything behind a single "getMessageSiblings" method that return the value of the "[$MESSAGE_SIBLINGS$]" variable. Ideally, MBMessageLocalService should simply invoke one method in this class for each of the variables it needs (right now the logic is spread across both classes).